### PR TITLE
1. Add Todo

### PR DESCRIPTION
**Pull request by hhe-wtag**
Adding todo using DOM, Vanilla JS

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -9,7 +9,20 @@
     <header id="header">
       <h1>The Greate Todo List 📝</h1>
```
Comment (hhu-wtag) [inline index.html]: typo
```suggestion
      <h1>The Great Todo List 📝</h1>
```

File: script.js [hunk 2]
```diff
@@ -1 +1,19 @@
+const todoInput = document.getElementById("todo-input");
+const addTodoBtn = document.getElementById("add-todo-btn");
+const taskListContainer = document.getElementById("task-list-container");
+const taskList = document.getElementById("task-list");
 
+addTodoBtn.addEventListener("click", addTodo);
+
+/**
+ * Adding todo, clearing input after adding
+ */
```
Comment (hhu-wtag) [inline script.js]: While comments can be helpful for future maintainers/collaborators, it's often a good practice to leave them out. This is also known as code smell. 
Ideally, your variable naming, function naming and code flow should be self describing. 

Let's remove this comment.
```suggestion
```

File: script.js [hunk 3]
```diff
@@ -1 +1,19 @@
+const todoInput = document.getElementById("todo-input");
+const addTodoBtn = document.getElementById("add-todo-btn");
+const taskListContainer = document.getElementById("task-list-container");
+const taskList = document.getElementById("task-list");
 
+addTodoBtn.addEventListener("click", addTodo);
```
Comment (hhu-wtag) [inline script.js]: The function names of event listeners should be prefixed with `on` or `handle` follwing the conventions.

```suggestion
addTodoBtn.addEventListener("click", onAddTodo);
```

File: script.js [hunk 4]
```diff
@@ -1 +1,19 @@
...
+ */
+function addTodo() {
+  const newTodo = document.createElement("li");
+  newTodo.innerHTML = todoInput.value;
+  const editIcon = document.createElement("span");
+  newTodo.style.listStyleType = "circle";
+  taskList.append(newTodo);
+
+  todoInput.value = "";
+}
```
Comment (hhu-wtag) [inline script.js]: 1. Rename as suggested
2. This function is called when the add button is clicked. So it should deligate the works that needed to be done to create a todo to other functions
```suggestion
let todos = [] // this list will hold todo objects

function renderTodos(){
  const todoTitle = todoInput.value
  const $todo = createTodo(todoTitle)
 // render todos list here by iterating over the todos array
}

function onAddTodo() {
  renderTodos()
  
  todoInput.value = "";
}

function createTodo(todoTitle){
  const $todo = document.createElement("li");
  
  $todo.innerHTML = todoInput.value;
  $todo.style.listStyleType = "circle";
  
  return $todo
}
```

**Review by hhu-wtag — CHANGES_REQUESTED** [review]

File: script.js (hunk 4 above)
Comment (hhe-wtag) [reply script.js]: Thank you for the suggestion bhaia, I have refactored the code to follow your suggested structure. Kindly review again.

File: script.js [hunk 5]
```diff
@@ -1 +1,38 @@
...
+
+let todos = [];
+
+function handleAddTodo() {
+  const newTodo = {
+    id: Date.now(),
+    title: $todoInput.value,
+    createdAt: Date.now().toLocaleString(),
+  };
+  if (newTodo.title.trim() === "") return;
```
Comment (hhu-wtag) [inline script.js]:
```suggestion
  };
  
  if (newTodo.title.trim() === "") return;
```

**Review by hhu-wtag — APPROVED** [review]
Nice work. Well Done.

**Pull request merged**